zebra-state/src/service.rs

```diff
@@ -29,14 +34,103 @@ struct StateService {
     /// Holds data relating to finalized chain state.
     sled: FinalizedState,
     /// Holds data relating to non-finalized chain state.
-    _mem: NonFinalizedState,
+    mem: NonFinalizedState,
+    /// Blocks awaiting their parent blocks for contextual verification.
+    contextual_queue: QueuedBlocks,
+}
+
+#[derive(Debug, Error)]
+#[error("block is not contextually valid")]
+struct CommitError(#[from] ValidateContextError);
+
+#[derive(displaydoc::Display, Debug, Error)]
+enum ValidateContextError {
+    /// block.height is lower than the current finalized height
+    OrphanedBlock,
 }
 
 impl StateService {
+    const REORG_LIMIT: usize = 100;
```

Comment:
This constant already exists:
```suggestion
    use crate::constants::MAX_BLOCK_REORG_HEIGHT;
```

Reply:
Actually, do you think we should move those constants to zebra-chain?

We use these constants in zebra-state and zebra-utils:
* MIN_TRASPARENT_COINBASE_MATURITY in zebra-state for contextual transaction verification
* MAX_BLOCK_REORG_HEIGHT in zebra-state for finalization, and in zebra-utils/zebra-checkpoints for the checkpoint range

Or we could make zebra-utils depend on zebra-state, but that seems like a waste for one constant. That said, we don't build it that often, so maybe it's better to have the good code structure?

Reply:
I still want it to be an associated constant but I've gone ahead and changed it so its derived from the shared constant which we can move around as needed.